arbor/communication/communicator.cpp

```diff
@@ -147,7 +149,12 @@ void communicator::update_connections(const recipe& rec,
     std::size_t n_con = 0;
     std::vector<std::vector<connection>> connections_by_src_domain(num_domains_);
     target_resolver.clear();
+    auto my_rank = ctx_->distributed->id();
+    std::unordered_map<cell_gid_type, std::unordered_set<cell_size_type>> outgoing_remote_targets;
```

Comment:
Two notes here:
1. For better or worse 'remote' has been used for co-simulation partners elsewhere, ie a separate program we exchange spikes with. To avoid confusion with those items I'd suggest a different name here.
2. Using a `set` is likely not the most frugal choice in terms of iteration speed and memory consumption, a vector should fit the usage patterns a bit better.